aiohttp/multipart.py

```diff
@@ -310,13 +311,14 @@ def _read_chunk_from_stream(self, size):
             self._prev_chunk = yield from self._content.read(size)
 
         chunk = yield from self._content.read(size)
-
+        self._content_eof += int(self._content.at_eof())
+        assert self._content_eof < 3, "Reading after EOF"
```

Comment:
The problem is also that boundary _starts with_ `\r\n`, and `maybe_readline` would add it to previous line/chunk. Backward lookup is also required.